lore/models.py

```diff
@@ -72,8 +73,13 @@ def hyper_parameter_search(
         """
         if scoring is None:
             scoring = None
+        if isinstance(self.estimator, SKLearn):
```

Comment:
feels like we have several checks like this now, it may make sense to subclass Model for each type soon